### PR TITLE
docs: cleanup, deprecate APIs

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

File: doc/lspconfig.txt [hunk 1]
```diff
@@ -129,50 +87,6 @@ All `config` elements described in `:help vim.lsp.start_client()` can
 additionally be overridden via the `setup {}` call. The most commonly
 passed overrides to `setup {}` are:
 
-- {capabilities} `table <string, string|table|bool|function>`
```
Comment (justinmk) [inline doc/lspconfig.txt:132]: don't repeat docs that already exist in `:help vim.lsp.start_client()`

File: doc/lspconfig.txt [hunk 2]
```diff
@@ -129,50 +87,6 @@ All `config` elements described in `:help vim.lsp.start_client()` can
...
-- {on_attach} `function(client, bufnr)`
-
-  Callback invoked by Nvim's built-in client when attaching a buffer to a
-  language server. Often used to set Nvim (buffer or global) options or to
-  override the Nvim client properties (`server_capabilities`) after a
-  language server attaches.
-
-  Prefer using an |LspAttach| autocommand handler instead.
-
 - {settings} `table <string, string|table|bool>`
```
Comment (justinmk) [inline doc/lspconfig.txt:90]: except this one looks rather useful. probably should move this into `:help lsp` ...

File: doc/lspconfig.txt [hunk 3]
```diff
@@ -328,10 +243,8 @@ ADDING NEW SERVERS                                    *lspconfig-new*
...
 The steps for adding and enabling a new server configuration are:
 
-1. load the `lspconfig` module (note that this is a stylistic choice) >
+1. Define the configuration (see also |vim.fs.root()|): >lua
    local lspconfig = require 'lspconfig'
-<
-2. define the configuration >
    local configs = require 'lspconfig.configs'
 
    -- Check if the config is already defined (useful when reloading this file)
```
Comment (justinmk) [inline doc/lspconfig.txt:250]: how to replace the `find_git_ancestor` call below with `vim.fs.root` ?

File: doc/lspconfig.txt [hunk 4]
```diff
@@ -500,9 +404,9 @@ is typically (in rough order):
 - the built-in client in Nvim core
 - nvim-lspconfig
 
-The first step in debugging is to test with a minimal configuration (such as
-`../test/minimal_init.lua`). Historically, many users problems are due to
-plugins or misconfiguration.
+The first step in debugging is to test with a minimal configuration:
+https://github.com/neovim/neovim/issues/new?assignees=&labels=bug%2Clsp&template=lsp_bug_report.yml(such as
```
Comment (justinmk) [inline doc/lspconfig.txt:408]: Link to core LSP bug report, which provides a minimal init.

Remove `minimal_init.lua` from this repo.

Comment (stucash) [reply doc/lspconfig.txt:408]: Hi, I am trying to set up my lsp first time in neovim and I have to be honest, it is not straightforward to see a default key mappings for neovim lsp, any where, until I found this `minimal_init.lua`. And you the link in `lspconfig.txt`, is a link to the bug report creation page, not a link to the `minimal_init` in core. If we could point to the right link for a minimal setup that'll be much appreciated. Personally,  I think it does make sense to make it a README must-have to mention where's the minimal default key maps for any lsp. Thanks very much!

Comment (stucash) [reply doc/lspconfig.txt:408]: > Link to core LSP bug report, which provides a minimal init.
> Hi, I am trying to set up my lsp first time in neovim and I have to be honest, it is not straightforward to see a default key mappings for neovim lsp, any where, until I found this `minimal_init.lua`. And you the link in `lspconfig.txt`, is a link to the bug report creation page, not a link to the `minimal_init` in core. If we could point to the right link for a minimal setup that'll be much appreciated. Personally,  I think it does make sense to make it a README must-have to mention where's the minimal default key maps for any lsp. Thanks very much!
> Remove `minimal_init.lua` from this repo.

Comment (stucash) [reply doc/lspconfig.txt:408]: > Link to core LSP bug report, which provides a minimal init.
> 
> Remove `minimal_init.lua` from this repo.



Hi, I am trying to set up my lsp first time in neovim and I have to be honest, it is not straightforward to see a default key mappings for neovim lsp, any where, until I found this `minimal_init.lua`. And you the link in `lspconfig.txt`, is a link to the bug report creation page, not a link to the `minimal_init` in core. If we could point to the right link for a minimal setup that'll be much appreciated. Personally,  I think it does make sense to make it a README must-have to mention where's the minimal default key maps for any lsp. Thanks very much!

Comment (justinmk) [reply doc/lspconfig.txt:408]: > whether default key maps should be part of lspconfig

Nvim 0.11 added default mappings for LSP. It's not released yet but you can try it here: https://github.com/neovim/neovim/releases

**Pull request merged**